cypress-E2E-tests/cypress/integration/asset_properties_widget.cy.ts

```diff
@@ -0,0 +1,614 @@
+  it('TC_Asset_Properties_Widget_config_029', () => {
+    const assetName = 'eee';
+    const searchResultsElement = '.legend > span';
+    const strongTextElement = '.c8y-empty-state > .d-flex > .text-medium';
+    const smallTextElement = '.small > small';
+    cy.get(searchElement).type(assetName);
+    cy.get(searchResultsElement).should('have.text', 'Search results');
+    cy.get(strongTextElement).should('have.text', 'No match found.');
+    cy.get(smallTextElement).should('have.text', 'Try to rephrase your search word.');
+  });
+
+  // Verify that on click of an asset in Asset selection section it should navigate to the next screen displaying empty state message
+  it('TC_Asset_Properties_Widget_config_030', () => {
+    const assetName = 'check 3';
+    const backArrowElement = '.dlt-c8y-icon-angle-left';
+    const filterElement = '.dlt-c8y-icon-filter';
+    cy.get(assetsTextElement).should('contain.text', 'Assets');
+    cy.get(asset_properties_widget_elements.loadMoreButton).click();
+    cy.get(`p[title='${assetName}']`).click();
+    cy.get(textElement).should('contains.text', 'Assets > ' + assetName);
+    cy.get(backArrowElement).should('exist');
+    cy.get('input[placeholder="Filter this column…"]').click();
+    cy.get(filterElement).should('exist');
+    cy.get('.c8y-empty-state > .d-flex > .text-medium').should(
+      'have.text',
+      'No results to display.'
+    );
+    cy.get('p.small').should('have.text', 'The selected asset has no children.');
+  });
+
+  // Verify that unchecking the property reflects under Properties section
+  it('TC_Asset_Properties_Widget_config_031', () => {
+    const checkboxSelectedElement = ':nth-child(1) > [style="width: 10%;"] > .ng-untouched';
+    const checkboxUnSelectElement = ':nth-child(1) > [style="width: 10%;"] > .ng-valid';
+    cy.get(checkboxSelectedElement).click();
+    cy.get(checkboxUnSelectElement).should('exist');
+  });
+
+  // Verify that cancel button works as expected
+  it('TC_Asset_Properties_Widget_config_032', () => {
+    cy.get(asset_properties_widget_elements.cancelButton).click();
+    cy.get(asset_properties_widget_elements.addWidgetButton).should('exist');
+  });
+
+  // Verify the presence of cancel and save button.
+  it('TC_Asset_Properties_Widget_config_033', () => {
+    cy.get(asset_properties_widget_elements.cancelButton).should('exist');
+    cy.get(asset_properties_widget_elements.saveButton).should('exist');
+  });
+
+  // Asset property widget title can be duplicate
+  it('TC_Asset_Properties_Widget_config_034', () => {
+    cy.selectAssetAndSave(assetName);
+    cy.get(asset_properties_widget_elements.widgetDashboardAddWidgetButton).click();
+    cy.get(asset_properties_widget_elements.cardElement).click();
+    cy.selectAssetAndSave(assetName);
+    cy.deleteWidgetInstances([assetProperties, assetProperties]);
+  });
+
+  // Verify the presence "No widgets to display" meessage in the dashboard
```

Comment:
Asset_Properties_Widget_view tests can be grouped  separately.

Reply:
again I have to repeat the before function so added within this